tidy3d/components/medium.py

```diff
@@ -539,10 +563,48 @@ def eps_diagonal_on_grid(
         ]
         return tuple(eps_list)
 
+    def eps_comp_on_grid(
+        self,
+        row: Axis,
+        col: Axis,
+        frequency: float,
+        coords: Coords,
+    ) -> Numpy:
```

Comment:
let's use `ArrayLike` in `types.py`. Do we know the number of dimensions and dtype? For example, if a 3D complex, can use `ArrayComplex3D` defined [here](https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/types.py#L109).

Change return docstring type too.

Reply:
fixed